### PR TITLE
Document how to backport a change into a release channel

### PR DESCRIPTION
**Pull request by curiousleo**
###### Motivation for this change

I keep forgetting to use `git cherry-pick -x`. So I wrote it down.

**[Preview here.](https://github.com/curiousleo/nixpkgs/blob/patch-2/.github/CONTRIBUTING.md#backporting-changes)**

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ In addition to writing properly formatted commit messages, it's important to inc
 
 For package version upgrades and such a one-line commit message is usually sufficient.
 
+## Backporting changes
+
+To backport a change into a release channel, take note of the commit in which the change was introduced into `master`. Then check out the release channel and use `git cherry-pick -x <original commit>`. When opening your backport PR, make sure to select the release channel branch as the target branch of the PR, and link to the PR in which the original change was made to `master`.
```
Comment (LnL7) [inline .github/CONTRIBUTING.md]: Perhaps also clarify what the release branches look like, pull requests should target eg. `release-19.09` and not the channel (`nixos-19.09`/`nixpkgs-unstable`) branches directly.

Comment (LnL7) [reply .github/CONTRIBUTING.md]: Also see https://nixos.org/nixpkgs/manual/#submitting-changes-stable-release-branches.

Comment (curiousleo) [reply .github/CONTRIBUTING.md]: Thanks! I've added a link and an example.

**Comment by curiousleo** [comment]
PTAL

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -48,6 +48,10 @@ In addition to writing properly formatted commit messages, it's important to inc
 
 For package version upgrades and such a one-line commit message is usually sufficient.
 
+## Backporting changes
+
+To [backport a change into a release branch](https://nixos.org/nixpkgs/manual/#submitting-changes-stable-release-branches), take note of the commit in which the change was introduced into `master`. Then check out the release branch (e.g. `release-19.09`, _not_ the channel) and use `git cherry-pick -x <original commit>`. When opening your backport PR, make sure to select the release branch as the target branch of the PR, and link to the PR in which the original change was made to `master`.
```
Comment (grahamc) [inline .github/CONTRIBUTING.md]: `_not_ the channel` is a bit too vague. If they don't know a release channel is named `release-19.09`, they probably won't know a channel is named `nixos-*` or `nixpkgs-*`. Add these two as examples of branches to not PR in to?

Comment (curiousleo) [reply .github/CONTRIBUTING.md]: Done.

**Review by nh2 — APPROVED** [review]
Looks sensible and clear to me.

**Review by vcunat — APPROVED** [review]

**Comment by worldofpeace** [comment]
Thank you soooo much for doing this @curiousleo :sparkles:.
Been on my todo list for a minute.

Though it's still frustrating that we have multiple documents and we have to veer away from the nixpkgs manual.

**Comment by curiousleo** [comment]
> Thank you soooo much for doing this @curiousleo 

My pleasure! I figured that after having to ask what the right `cherry-pick` incantation was I'd write it down :)

> Though it's still frustrating that we have multiple documents and we have to veer away from the nixpkgs manual.

Yeah

**Pull request merged**